Source/Initialization/WarpXInitData.cpp

```diff
@@ -105,6 +105,169 @@ WarpX::PostProcessBaseGrids (BoxArray& ba0) const
     }
 }
 
+void
+WarpX::PrintMainPICparameters ()
+{
+    amrex::Print()<<"-------------------------------------------------------------------------------"<<"\n";
+    amrex::Print()<<"--------------------------- MAIN EM PIC PARAMETERS ----------------------------"<<"\n";
+    amrex::Print()<<"-------------------------------------------------------------------------------"<<"\n";
+
+    // Print geometry dimensionality
+    amrex::ParmParse pp_geometry("geometry");
+    std::string dim;
+    pp_geometry.query( "dims", dim );
```

Comment:
Here I would match the variable name `dim` with the name of the input parameter `dims`. This is usually less confusing, especially for those who may `git grep "dims"` in the repository:
```suggestion
    std::string dims;
    pp_geometry.query( "dims", dims );
```